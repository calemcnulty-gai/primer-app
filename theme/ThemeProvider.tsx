import React, { createContext, useContext, ReactNode } from 'react';
import { Colors } from '@/constants/Colors';

// Create context with colors only
type ThemeContextType = {
  colors: typeof Colors;
};

const ThemeContext = createContext<ThemeContextType | undefined>(undefined);

// Theme provider props
type ThemeProviderProps = {
  children: ReactNode;
};

export const ThemeProvider = ({ children }: ThemeProviderProps) => {
  return (
    <ThemeContext.Provider
      value={{
        colors: Colors,
      }}
    >
      {children}
    </ThemeContext.Provider>
  );
};

// Custom hook to use the theme
export const useTheme = () => {
  const context = useContext(ThemeContext);
  if (context === undefined) {
    throw new Error('useTheme must be used within a ThemeProvider');
  }
  return context;
}; 