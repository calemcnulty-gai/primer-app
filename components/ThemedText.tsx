import { Text, type TextProps, StyleSheet } from 'react-native';
import { useTheme } from '@/theme/ThemeProvider';

export type ThemedTextProps = TextProps & {
  customColor?: string;
  type?: 'default' | 'title' | 'defaultSemiBold' | 'subtitle' | 'link';
};

export function ThemedText({
  style,
  customColor,
  type = 'default',
  ...rest
}: ThemedTextProps) {
  const { colors } = useTheme();
  
  // Use provided color, otherwise use theme color
  const textColor = customColor || colors.text;

  return (
    <Text
      style={[
        { color: textColor },
        type === 'default' ? styles.default : undefined,
        type === 'title' ? styles.title : undefined,
        type === 'defaultSemiBold' ? styles.defaultSemiBold : undefined,
        type === 'subtitle' ? styles.subtitle : undefined,
        type === 'link' ? [styles.link, { color: colors.tint }] : undefined,
        style,
      ]}
      {...rest}
    />
  );
}

// Fonts styled to look good with the e-ink, paper-like theme
const styles = StyleSheet.create({
  default: {
    fontSize: 16,
    lineHeight: 24,
    fontFamily: 'SpaceMono',
  },
  defaultSemiBold: {
    fontSize: 16,
    lineHeight: 24,
    fontWeight: '600',
    fontFamily: 'SpaceMono',
  },
  title: {
    fontSize: 28,
    fontWeight: 'bold',
    lineHeight: 32,
    fontFamily: 'SpaceMono',
  },
  subtitle: {
    fontSize: 20,
    fontWeight: 'bold',
    fontFamily: 'SpaceMono',
  },
  link: {
    lineHeight: 24,
    fontSize: 16,
    textDecorationLine: 'underline',
    fontFamily: 'SpaceMono',
  },
});
