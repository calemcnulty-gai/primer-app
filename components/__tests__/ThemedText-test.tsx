import * as React from 'react';
import renderer from 'react-test-renderer';

import { ThemedText } from '../ThemedText';
import { ThemeProvider } from '@/theme/ThemeProvider';

it(`renders correctly`, () => {
  const tree = renderer.create(
    <ThemeProvider>
      <ThemedText>Snapshot test!</ThemedText>
    </ThemeProvider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
