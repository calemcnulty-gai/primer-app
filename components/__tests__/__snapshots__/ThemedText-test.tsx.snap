// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#5b4636",
      },
      {
        "fontFamily": "SpaceMono",
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;
